File: Dart/fibonacci.dart
/*
* Recursive version of fibonacci series
* [Language used]: Dart 2.9.2
* Author: @VarunSAthreya
*/

// Recursive fibonacci number generating method
int fibonacci(int n) {
  if (n <= 1) {
    // base case
    return 1;
  } else {
    // recursive call
    return fibonacci(n - 2) + fibonacci(n - 1);
  }
}

// main method
main() {
  int _n = 10;

  // print first n fibonacci numbers
  for (int i = 0; i < _n; ++i) {
    print(fibonacci(i));
  }
}

File: lua/fibonacci.lua
print("Fibonacci series up to 10 term")

n1 = 0
n2 = 1
nextTerm = 0
print(1)
for i = 1, 9, 1
	do
		nextTerm = n1 + n2
		print(nextTerm)
		n1 = n2
		n2 = nextTerm
	end
